### ISSUE TITLE
Allow options to consider c5 and c4 instances equal

### ISSUE THREAD

**Issue by binarylogic** [cristim/autospotting#206]
# Github issue #

## Issue type ##

- Feature Idea

## Summary ##

When a `c5.*` instance is specified as the on-demand instance, `c4.*` instances are never used because they contain 500mb less than their `c5` counterpart. The autospotting algorithm excludes `c4` instances it due to this memory difference.

## Expected results ##

This _is_ technically correct, and I'm on the fence if this should be the default behavior, but most would expect a `c4` instance to be a valid replacement for a `c5` instance.

**Comment by cristim** [comment]
Good catch, we can relax the memory comparison algorithms to accept a small difference if people expect this.

The  problem is that tose 500M are pretty much negligible for larger instance types but this is not okay for the smallest micro and nano instances where this makes a lot of difference.

We could make it a default as long as we carefully set it to not cover the smaller instances. Maybe accept a difference in both percentage(up to 15-20%) and absolute value of 500M, whichever is smaller.

**Comment by binarylogic** [comment]
Yep, that was exactly my thought. `c5.*` instances offer exactly 93.75% of the memory that `c4.*` instances have available. I'm thinking we allow an even 10% difference.

**Comment by cristim** [comment]
Looking forward for a PR addressing this

**Comment by russellballestrini** [comment]
I can duplicate this issue, but `c4` also is not replaced by `c4` so I think there is a bigger defect at the root.

**Comment by russellballestrini** [comment]
This is still an issue but as a work around you can use the `allowed_instance_type` option to pass a list of valid instance types. I have verified that this works as a work around.

  https://github.com/cristim/autospotting/blob/master/autospotting.go#L121-L125

**Comment by gabegorelick** [comment]
Similar issue happens for EBS, which is mentioned in #353. If you have a t3 instance, t2 will not be considered since it lacks EBS throughput. This will probably continue to be an issue in the future as newer instance types add new features that older instances types will naturally lack. 

I think educating users about `autospotting_allowed_instance_types` is a fine solution though.

**Comment by cristim** [comment]
For EBS we now have a flag that makes the comparison optional.

The issue with C4 and C5 is the memory size difference. This should be easy to fix, although nowadays few people use C3 and older instances anymore.

@gabegorelick @russellballestrini, @binarylogic: are you guys still interested in this?
